concourse/model/traits/update_component_deps.py

```diff
@@ -159,6 +159,29 @@ def merge_method(self):
         doc='whether or not created PRs should be automatically merged. **deprecated**',
         type=MergePolicy,
     ),
+    AttributeSpec.optional(
+        name='additional_ctx_repositories',
```

Comment:
naming: I think we should refer to `ocm_repositories` for new code ([ocm-spec](https://github.com/open-component-model/ocm-spec/blob/main/doc/introduction/component_repository.md)).

semantics: let us rather define the contract like so:

- if no ocm-repositories are configured at all, use fallback to current behaviour (as you described in pull-request-description); if feasible, preferably return a a generated cfg (we use the same pattern for other traits)
- if ocm-repositories are configured, they are _the_ cfg to consider (no fallback)

Considering structure, consider OCM-CLI's spec:

https://github.com/open-component-model/ocm/blob/main/docs/reference/ocm_configfile.md
-> ocm.config.ocm.software (resolvers, ..)

Also, consider factoring the model-class out of the `concourse` package. ideally, this should go to package `ocm` (which does not exist yet), next-best package: `cnudie`?